### PR TITLE
[Chore] Optimize swift programming exercise

### PR DESCRIPTION
**Pull request by derLalla**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

We want to update the Swift programming exercise itself and we also want to update the available test cases of Swift programming exercises.

### Description

This adds pseudo structural tests to the Swift programming exercise. Pseudo, because we just check if an attribute or method exists and do not test the actual functionality. Using reflection in Swift has proven to be quite not feasible without providing too much information to the student`s template repository. We would need to provide an interface which already contains all needed functions to be able to use reflection during the tests.

We also updated the Dockerfile for a local Swift setup. Which now uses the latest swift version and simplifies the process.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a Swift Exercise
4. Test that the template repository has a 0% score and no passing test cases
5. Test that the solution repository has a 100% score and all tests are passing
6. Cause the build to fail and find a `red x` in the online editor

### Test Coverage

Stays the same.

#### Failed Build indicator:
image
#### New test cases:
image

### CODE REVIEW

File: src/main/docker/bamboo/swift/Dockerfile [hunk 1]
```diff
@@ -17,28 +17,13 @@ RUN update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/jav
 # Add file in /artemis/bin/mvn that uses the correct java version and passes all arguments to mvn
 RUN mkdir /artemis && mkdir /artemis/bin && printf '#!/bin/bash\nJAVA_HOME=/usr/lib/jvm/java-15-oracle /usr/bin/mvn "$@"\n' > /artemis/bin/mvn && chmod 777 /artemis/bin/mvn
 
-# Install Swift and SwiftLint (Adapted from norionomura/swift and norionomura/swiftlint)
-ENV SWIFT_BRANCH swift-5.3.1-release
-ENV SWIFT_PLATFORM ubuntu18.04
-ENV SWIFT_VERSION 5.3.1-RELEASE
+# Install Swift by copying relevant files
+COPY --from=swift:latest /usr /usr
```
Comment (derLalla) [inline src/main/docker/bamboo/swift/Dockerfile:21]: this is standard, see: https://docs.docker.com/develop/develop-images/multistage-build/#use-an-external-image-as-a-stage

**Review by tiffca — DISMISSED** [review]
Tested successfully.

Note the first initial build is a little slow, so might need to give it some time. 

Also, for the red x to show, try refreshing the page after making an error. So, in the code editor for the template programming exercise, cause a build to fail as in the example, submit, then refresh the page once the build is done.

**Review by tiffca — APPROVED** [review]
Re-approve

**Comment by SimonEntholzer** [comment]
Code looked good as well :+1:

**Review by SimonEntholzer — APPROVED** [review]
Code looked good too, tested with @ivanchimeno on TS1 :+1:

**Review by ivanchimeno — APPROVED** [review]

**Pull request merged**